parser/parser_test.go

```diff
@@ -673,3 +676,81 @@ func TestParseMultiByte(t *testing.T) {
 		})
 	}
 }
+
+func TestCreateRequest(t *testing.T) {
+	cases := []struct {
+		input    string
+		expected *api.CreateRequest
+	}{
+		{
+			`FROM test`,
+			&api.CreateRequest{From: "test"},
+		},
+		{
+			`FROM test
+TEMPLATE some template
+`,
+			&api.CreateRequest{
+				From:     "test",
+				Template: "some template",
+			},
+		},
+		{
+			`FROM test
+LICENSE single license
+PARAMETER temperature 0.5
+MESSAGE user Hello
+`,
+			&api.CreateRequest{
+				From:       "test",
+				License:    []string{"single license"},
+				Parameters: map[string]any{"temperature": float32(0.5)},
+				Messages: []api.Message{
+					{Role: "user", Content: "Hello"},
+				},
+			},
+		},
+		{
+			`FROM test
```

Comment:
Possible/worth adding some tests from local files?

Reply:
We can, but there is a *lot* of coverage for this already in the various `routes_*_test.go` so any test here would be pretty redundant.